crates/distribution-types/src/lib.rs

```diff
@@ -166,6 +167,10 @@ pub struct DirectUrlBuiltDist {
     /// We require that wheel urls end in the full wheel filename, e.g.
     /// `https://example.org/packages/flask-3.0.0-py3-none-any.whl`
     pub filename: WheelFilename,
+    /// The url without subdirectory fragment.
+    pub location: Url,
+    pub subdirectory: Option<PathBuf>,
+    /// The url with subdirectory fragment.
```

Comment:
The intention is to eventually remove the `VerbatimUrl`, leaving only `given: String`, for now we just eliminate the fallible re-parsing of urls